Week3/homework/step2-4.js

```diff
@@ -1,7 +1,9 @@
 'use strict';
 
 function Dog() {
-  // add your code here
+  this.name = 'Karabas';
+  this.color = 'Black and white';
+  this.numLegs = 4;
```

Comment:
Ah. This is not completely right. You are correct in the way of creating a constructor function however, the values that you have provided to it i.e `Karabas `, `Black and white` and `4` are all hardcoded. This means, I can't use this constructor function to create a new dog object. The right way of doing it will be using function parameters:
```
function Dog(name, color, legs) {
  this.name = name;
  this.color = color;
  this.numLegs = legs;
}
```

And now you can use this constructor to create any number of `dog` objects as you like.

```
const one = new Dog('Caramel`, 'Beige', 4);
const two = new Dog('Georgie', 'Black', 4);
// and so on
```

Reply:
I think I did this on freecodecamp with my solution and it worked, so I didn't put much thought into it. But you are right, the whole point of creating a constructor function is to be able to pass whichever argument we want. Thank you!